2020/3/trees/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+use std::fs::File;
+use std::io::{self, BufRead};
+
+pub struct Map {
+    trees: Vec<String>
+}
+
+impl Map {
+    pub fn parse_file(path: &str) -> Map {
+        let file = File::open(path);
+    
+        let reader = io::BufReader::new(file.unwrap());
+        let mut trees = Vec::new();
+        for line in reader.lines() {
+            trees.push(line.unwrap());
+        }
+
+        Map::parse_lines(trees)
+    }
+
+    pub fn parse_string(map: &str) -> Map {
+        let mut trees = Vec::new();
+        for line in map.split_whitespace() {
+            trees.push(String::from(line));
+        }
+        Map::parse_lines(trees)
+    }
+
+    fn parse_lines(map: Vec<String>) -> Map {
+        Map { trees: map }
+    }
+
+    pub fn is_tree(&self, row: usize, col: usize) -> bool {
+        let len = self.trees[0].len();
+        let c = col % len;
+        self.trees[row].chars().nth(c).unwrap() == '#'
+    }
+
+    pub fn count_trees(&self, right: i32, down: i32) -> i32 {
```

Comment:
It's generally idiomatic to use `isize`/`usize` unless you're dealing with a network protocol, file format, tight memory encoding, etc and absolutely need to specify the bit length.

It's more efficient to just carry around integers of native-word-size, since often the others are just going to get aligned to that size anyway.

Reply:
Fixed in 459565e47de7c8d766302a246a7948fee3050740